packages/apputils-extension/src/index.ts

```diff
@@ -90,10 +91,60 @@ const paletteRestorer: JupyterFrontEndPlugin<void> = {
 const settings: JupyterFrontEndPlugin<ISettingRegistry> = {
   id: '@jupyterlab/apputils-extension:settings',
   activate: async (app: JupyterFrontEnd): Promise<ISettingRegistry> => {
-    const connector = app.serviceManager.settings;
-    const plugins = (await connector.list()).values;
+    const { isDeferred, isDisabled } = PageConfig.Extension;
+    const connector = new (class SettingConnector extends DataConnector<
+      ISettingRegistry.IPlugin,
+      string
+    > {
+      fetch(id: string): Promise<ISettingRegistry.IPlugin> {
+        return app.serviceManager.settings.fetch(id);
+      }
+
+      async list(
+        query: 'active' | 'all' = 'all'
```

Comment:
Would it make sense to move this functionality to the `services` package itself? It seems as if it might have reuse value? That might also formalize it more, as we can add a docstring explaining that `active` implies that it is not disabled, and not deferred.

Reply:
I thought about putting it in services, but it seemed very specific to the functionality of the `SettingRegistry` and it seemed to me that the services package should be *completely* agnostic about what you're doing with the data being returned from the back-end.

Another place I considered putting this logic was as a query parameter to the back-end, which still remains a possibility (in which case, I'd agree with your suggestion to move this into services).

But for now, this seemed like the most unobtrusive thing to do because it seemed to localize the changes to the actual location of the concern. Do you disagree?